pkg/process/monitor/process_monitor_test.go

```diff
@@ -75,6 +80,35 @@ func TestProcessMonitorSanity(t *testing.T) {
 		return numberOfExecs.Load() > 1
 	}, time.Second, time.Millisecond*200, "didn't capture exec events %d", numberOfExecs.Load())
 
+	tel := telemetry.ReportPrometheus()
+	getCounter := func(m string) float64 {
+		pm, ok := tel[m]
+		if !ok {
+			return 0
+		}
+		switch v := pm.(type) {
+		case libtelemetry.Counter:
+			return v.WithValues().Get()
+		case libtelemetry.Gauge:
+			return v.WithValues().Get()
+		}
+		return 0
+	}
```

Comment:
Can you please re-write the tests following my previous suggestion of relying on the public API of the library?

Example:
```
assert.Equal(t, pm.telemetry.metricName.Get(), 1)
```

Not only the tests are unnecessarily complex at the moment, but they are completely coupled to Prometheus, and one of the goals of the `telemetry` library is precisely abstracting away the details of how data gets reported. In other words, changing a metric from `Prometheus` to `PayloadTelemtry` should *not* break your tests.

Reply:
As discuss offline let's not test metrics end to end.
We could add a test in the telemetry test suite